packages/react-accordion/src/components/Accordion/useAccordionContext.ts

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import { createDescendantContext, useDescendant, useDescendantsInit } from '@reach/descendants';
+import { AccordionContext, AccordionDescendant, AccordionOpen, AccordionState } from './Accordion.types';
+import { useConst } from '@fluentui/react-utilities';
+
+export const accordionDescendantContext = createDescendantContext<AccordionDescendant>('AccordionDescendantContext');
+
+export const accordionContext = React.createContext<AccordionContext>(undefined!);
+
+export function useCreateAccordionContext(state: AccordionState) {
+  const { open, multiple, collapsible, onToggle } = state;
+  const isControlled = useConst(typeof open !== 'undefined');
+  const [descendants, setDescendants] = useDescendantsInit<AccordionDescendant>();
+  const [openItems, setOpenItems] = React.useState<AccordionOpen>(() => initializeOpenItems(state));
+
+  const requestToggle = React.useCallback(
+    (index: number) => {
+      onToggle?.(index);
+      if (!isControlled) {
+        setOpenItems(previousOpenItems =>
+          updateOpenItems(index, previousOpenItems, {
+            collapsible,
+            multiple,
+          }),
+        );
+      }
+    },
+    [collapsible, isControlled, multiple, onToggle],
+  );
+  const context = React.useMemo<AccordionContext>(
+    () => ({
+      openItems: isControlled ? open! : openItems,
+      requestToggle,
+    }),
+    [isControlled, open, openItems, requestToggle],
```

Comment:
We should consider to use there `react-context-selector` as `open` might be unstable and cause unnecessary re-renders for child components. Let's sync on it in offline.